Libraries/Components/TabBarIOS/TabBarItemIOS.ios.js

```diff
@@ -10,89 +10,101 @@
-     */
-    systemIcon: PropTypes.oneOf([
-      'bookmarks',
-      'contacts',
-      'downloads',
-      'favorites',
-      'featured',
-      'history',
-      'more',
-      'most-recent',
-      'most-viewed',
-      'recents',
-      'search',
-      'top-rated',
-    ]),
-    /**
-     * A custom icon for the tab. It is ignored when a system icon is defined.
-     */
-    icon: Image.propTypes.source,
-    /**
-     * A custom icon when the tab is selected. It is ignored when a system
-     * icon is defined. If left empty, the icon will be tinted in blue.
-     */
-    selectedIcon: Image.propTypes.source,
-    /**
-     * Callback when this tab is being selected, you should change the state of your
-     * component to set selected={true}.
-     */
-    onPress: PropTypes.func,
-    /**
-     * If set to true it renders the image as original,
-     * it defaults to being displayed as a template
-     */
-    renderAsOriginal: PropTypes.bool,
-    /**
-     * It specifies whether the children are visible or not. If you see a
-     * blank content, you probably forgot to add a selected one.
-     */
-    selected: PropTypes.bool,
-    /**
-     * React style object.
-     */
-    style: DeprecatedViewPropTypes.style,
-    /**
-     * Text that appears under the icon. It is ignored when a system icon
-     * is defined.
-     */
-    title: PropTypes.string,
-    /**
-     *(Apple TV only)* When set to true, this view will be focusable
-     * and navigable using the Apple TV remote.
-     *
-     * @platform ios
-     */
-    isTVSelectable: PropTypes.bool,
-  };
+import type {ViewProps} from 'ViewPropTypes';
+import type {ColorValue} from 'StyleSheetTypes';
+import type {SyntheticEvent} from 'CoreEventTypes';
+import type {ImageSource} from 'ImageSource';
+
+type Props = $ReadOnly<{|
+  ...ViewProps,
+
+  /**
+   * Little red bubble that sits at the top right of the icon.
+   */
+  badge?: ?(string | number),
+
+  /**
+   * Background color for the badge. Available since iOS 10.
+   */
+  badgeColor?: ColorValue,
+
+  /**
+   * Items comes with a few predefined system icons. Note that if you are
+   * using them, the title and selectedIcon will be overridden with the
+   * system ones.
+   */
+  systemIcon?: ?(
+    | 'bookmarks'
+    | 'contacts'
+    | 'downloads'
+    | 'favorites'
+    | 'featured'
+    | 'history'
+    | 'more'
+    | 'most-recent'
+    | 'most-viewed'
+    | 'recents'
+    | 'search'
+    | 'top-rated'
+  ),
+
+  /**
+   * A custom icon for the tab. It is ignored when a system icon is defined.
+   */
+  icon?: ?ImageSource,
+
+  /**
+   * A custom icon when the tab is selected. It is ignored when a system
+   * icon is defined. If left empty, the icon will be tinted in blue.
+   */
+  selectedIcon?: ?ImageSource,
+
+  /**
+   * Callback when this tab is being selected, you should change the state of your
+   * component to set selected={true}.
+   */
+  onPress?: ?(event: SyntheticEvent<null>) => mixed,
```

Comment:
This might raise an eyebrow, so thought I'd explain reason I typed `event` like this; the `nativeEvent` for `onPress` is `nil` in the native code:

https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/React/Views/RCTTabBar.m#L203-L213